widgetssdk/src/main/java/com/glia/widgets/chat/controller/ChatController.java

```diff
@@ -1241,9 +1284,64 @@ public void singleChoiceOptionClicked(
         }
     }
 
-    public void sendCustomCardResponse(String text, String value) {
-        SingleChoiceAttachment attachment = new ResponseAttachment(value);
-        sendMessageUseCase.execute(text, attachment, sendMessageCallback);
+    public void sendCustomCardResponse(String messageId, String text, String value) {
+        sendMessageUseCase.execute(messageId, text, value, sendMessageCallback);
+
+        chatState.chatItems.stream()
+                .filter(chatItem -> messageId.equals(chatItem.getId()))
+                .map(chatItem -> (CustomCardItem) chatItem)
+                .findFirst()
+                .ifPresent(chatItem -> {
+                    Integer customCardType = customCardTypeUseCase.execute(chatItem.getViewType());
+                    if (customCardType == null) {
+                        return;
+                    }
+
+                    ChatMessage currentMessage = chatItem.getMessage();
+                    boolean showCustomCard = customCardShouldShowUseCase.execute(
+                            currentMessage,
+                            customCardType,
+                            false
+                    );
+
+                    List<ChatItem> chatItems = new ArrayList<>(chatState.chatItems);
+                    int indexInList = chatItems.indexOf(chatItem);
+                    if (showCustomCard) {
+                        indexInList += 1;
+                    } else {
+                        chatItems.remove(chatItem);
+                    }
```

Comment:
To me it makes sense to add some comments explaining the logic here: why do we increment the index, and why the item is being removed.

Reply:
Added the comments.